test/Core.Test/Services/GroupServiceTests.cs

```diff
@@ -30,12 +33,116 @@ public GroupServiceTests()
             );
         }
 
-        // Remove this test when we add actual tests. It only proves that
-        // we've properly constructed the system under test.
         [Fact]
-        public void ServiceExists()
+        public async Task SaveAsync_DefaultGroupId_CreatesGroupInRepository()
```

Comment:
It would be nice to have a happy path test of the Replace flow (where `id != default`) as well

Reply:
Added `SaveAsync_NonDefaultGroupId_ReplaceGroupInRepository` test